packages/core/test/insured-bridge/BridgePool.js

```diff
@@ -995,4 +1020,51 @@ describe("BridgePool", () => {
+    it("Deposit liquidity", async () => {
+      // Before adding liquidity pool should have none and LP should have no LP tokens.
+      assert.equal((await l1Token.methods.balanceOf(bridgePool.options.address).call()).toString(), "0");
+      assert.equal((await lpToken.methods.balanceOf(rando).call()).toString(), "0");
+
+      // Starting exchange rate should be 1, even though there is no liquidity in the pool.
+      assert.equal((await bridgePool.methods.exchangeRateCurrent().call()).toString(), toWei("1"));
+
+      // Approve funds and add to liquidity.
+      await l1Token.methods.approve(bridgePool.options.address, MAX_UINT_VAL).send({ from: rando });
+      await bridgePool.methods.addLiquidity(toWei("10")).send({ from: rando });
+
+      // Check balances have updated accordingly. Check liquid reserves incremented. Check exchange rate unchanged.
+      assert.equal((await l1Token.methods.balanceOf(bridgePool.options.address).call()).toString(), toWei("10"));
+      assert.equal((await lpToken.methods.balanceOf(rando).call()).toString(), toWei("10"));
+      assert.equal((await bridgePool.methods.liquidReserves().call()).toString(), toWei("10"));
+      assert.equal((await bridgePool.methods.utilizedReserves().call()).toString(), toWei("0"));
+      assert.equal((await bridgePool.methods.exchangeRateCurrent().call()).toString(), toWei("1"));
+    });
+    it("Withdraw liquidity", async () => {
+      // Approve funds and add to liquidity.
+      await l1Token.methods.approve(bridgePool.options.address, MAX_UINT_VAL).send({ from: rando });
+      await bridgePool.methods.addLiquidity(toWei("10")).send({ from: rando });
+
+      // LP redeems half their liquidity. Balance should change accordingly.
+      await bridgePool.methods.removeLiquidity(toWei("5")).send({ from: rando });
+
+      assert.equal((await l1Token.methods.balanceOf(bridgePool.options.address).call()).toString(), toWei("5"));
+      assert.equal((await lpToken.methods.balanceOf(rando).call()).toString(), toWei("5"));
+      assert.equal((await bridgePool.methods.liquidReserves().call()).toString(), toWei("5"));
+      assert.equal((await bridgePool.methods.utilizedReserves().call()).toString(), toWei("0"));
+      assert.equal((await bridgePool.methods.exchangeRateCurrent().call()).toString(), toWei("1"));
+
+      // LP redeems their remaining liquidity. Balance should change accordingly.
+      await bridgePool.methods.removeLiquidity(toWei("5")).send({ from: rando });
+
+      assert.equal((await l1Token.methods.balanceOf(bridgePool.options.address).call()).toString(), toWei("0"));
+      assert.equal((await lpToken.methods.balanceOf(rando).call()).toString(), toWei("0"));
+      assert.equal((await bridgePool.methods.liquidReserves().call()).toString(), toWei("0"));
+      assert.equal((await bridgePool.methods.utilizedReserves().call()).toString(), toWei("0"));
+      assert.equal((await bridgePool.methods.exchangeRateCurrent().call()).toString(), toWei("1"));
```

Comment:
note for now these tests dont validate any exchange rate functions. we will need the ability to finalize L2-L1 withdraws to fully capture this.